### PR TITLE
Add new package Dlib

### PR DESCRIPTION
**Pull request by bfovet**

### CODE REVIEW

File: var/spack/repos/builtin/packages/dlib/package.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    homepage = "http://dlib.net"
+    url      = "https://github.com/davisking/dlib/archive/v19.19.tar.gz"
+
+    version('19.19', sha256='7af455bb422d3ae5ef369c51ee64e98fa68c39435b0fa23be2e5d593a3d45b87')
+
+    variant('cuda', default=False, description='Build with CUDA')
+    variant('blas', default=True, description='Build with BLAS')
+    variant('lapack', default=True, description='Build with LAPACK')
+    variant('mkl', default=False, description='Build with the MKL library')
+    variant('avx', default=False, description='Use AVX instructions')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dlib/package.py]: @alalazo is this something that could use our better microarch support instead of a variant?

Comment (alalazo) [reply var/spack/repos/builtin/packages/dlib/package.py]: Yes. To be consistent with `fftw` and other packages we should remove the variant and add the corresponding configure option if `avx in self.target`.

File: var/spack/repos/builtin/packages/dlib/package.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    variant('shared', default=False, description='Build shared libraries')
+
+    depends_on('blas')
+    depends_on('lapack')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/dlib/package.py]: These dependencies are not conditional on the variants. Is this correct? If so, what would be the meaning of depending on e.g. `lapack` if `~lapack`?

**Comment by adamjstewart** [comment]
Ping @bfovet

**Comment by alalazo** [comment]
Closing as outdated. Feel free to reopen if you want to continue working on this.

**Pull request closed without merging**